src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -436,11 +436,11 @@ private function readProperty(array $zval, string $property, bool $ignoreInvalid
             }
         } catch (\Error $e) {
             // handle uninitialized properties in PHP >= 7.4
-            if (\PHP_VERSION_ID >= 70400 && preg_match('/^Typed property ([\w\\\]+)::\$(\w+) must not be accessed before initialization$/', $e->getMessage(), $matches)) {
-                $r = new \ReflectionProperty($matches[1], $matches[2]);
+            if (\PHP_VERSION_ID >= 70400 && preg_match('/^Typed property ('.preg_quote(get_debug_type($object), '/').')::\$(\w+) must not be accessed before initialization$/', $e->getMessage(), $matches)) {
```

Comment:
@nicolas-grekas
 
This change introduced a new bug:

```php
class X { public int $a; }
class Y extends X {}
(new Y)->a;
```

This throws: 

```
Typed property X::$a must not be accessed before initialization
```

but since `get_debug_type(new Y)` returns `Y` the error message does not match

```
Typed property Y::$a must not be accessed before initialization"
```

Think my original code would fix this.